logstash-core/spec/api/spec_helper.rb

```diff
@@ -11,9 +9,107 @@
 
 ENV['RACK_ENV'] = 'test'
 
-Rack::Builder.parse_file(File.join(ROOT, 'init.ru'))
+Rack::Builder.parse_file(File.join(API_ROOT, 'init.ru'))
 
 def read_fixture(name)
   path = File.join(File.dirname(__FILE__), "fixtures", name)
   File.read(path)
 end
+
+module LogStash
+  class DummyAgent < Agent
+    def fetch_config(settings)
+      "input { generator {count => 0} } output { null { } }"
+    end
+
+    def start_webserver; end
+    def stop_webserver; end
+  end
+end
+
+##
+# Class used to wrap and manage the execution of an agent for test,
+# this helps a lot in order to have a more integrated test for the
+# web api, could be also used for other use cases if generalized enought
+##
+class LogStashRunner
+
+  attr_reader :config_str, :agent, :pipeline_settings
+
+  def initialize
+    args = [
+      :logger => Cabin::Channel.get(LogStash),
+      :auto_reload => false,
+      :collect_metric => true,
+      :debug => false,
+      :node_name => "test_agent",
+      :web_api_http_port => rand(9600..9700)
+    ]
+
+    @config_str   = "input { generator {count => 0} } output { null { } }"
```

Comment:
I believe you can just do output {} to have an empty output section, this eliminating the need for the logstash-output-null